lib/geordi/gitlinear.rb

```diff
@@ -0,0 +1,201 @@
+require 'highline'
+require 'net/http'
+require 'json'
+require 'active_support/core_ext/object/blank'
+
+module Geordi
+  class Gitlinear
+    # This require-style is to prevent Ruby from loading files of a different
+    # version of Geordi.
+    require File.expand_path('settings', __dir__)
+
+    API_ENDPOINT = 'https://api.linear.app/graphql'.freeze
+
+    def initialize
+      self.highline = HighLine.new
+      self.settings = Settings.new
+    end
+
+    def run_commit(git_args)
+      Interaction.warn <<~WARNING unless Util.staged_changes?
+        No staged changes. Will create an empty commit.
+      WARNING
+
+      issue = choose_issue
+      if issue
+        create_commit "[#{issue['identifier']}] #{issue['title']}", "Issue: #{issue['url']}", *git_args
+      end
+    end
+
+    def run_branch(from_master: false)
+      issue = choose_issue || Interaction.fail('No issue selected.')
+
+      branch_list_string = if Util.testing?
+                             ENV['GEORDI_TESTING_GIT_BRANCHES'] || ''
+                           else
+                             `git branch --format="%(refname:short)"`
+                           end
+
+      if branch_list_string.nil? || branch_list_string.strip.empty?
+        Interaction.fail 'Could not determine local git branches.'
+      end
+
+      new_branch_name = "#{issue['branchName']}"
```

Comment:
Why not like this? We could even let go of this variable, `issue['branchName']` is perfect.
```suggestion
      new_branch_name = issue['branchName']
```